controllers/secret_controller.go

```diff
@@ -206,6 +217,39 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	}
 }
 
+// getUsernameAnno retrieves the username associated with a given node and ecrypts it using the given key
+func (r *SecretReconciler) getUsernameAnno(ctx context.Context, node core.Node, key []byte) (string, error) {
```

Comment:
As far as the function name, I'll change it to `getEncryptedUsername` for clarity. However, I think it should stay as part of the reconciler since it needs information about the configmap's namespace, and access to the configmap parsing which is in configmap_controller.